app/migrations/Version20200309172908.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sylius\Migrations;
+
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\Migrations\AbstractMigration;
+
+/**
+ * Auto-generated Migration: Please modify to your needs!
+ */
+final class Version20200309172908 extends AbstractMigration
+{
+    public function getDescription() : string
+    {
+        return '';
+    }
+
+    public function up(Schema $schema) : void
+    {
+        // this up() migration is auto-generated, please modify it to your needs
+        $this->abortIf($this->connection->getDatabasePlatform()->getName() !== 'mysql', 'Migration can only be executed safely on \'mysql\'.');
+
+        $this->addSql('ALTER TABLE sylius_product_variant ADD enabled TINYINT(1) NOT NULL');
```

Comment:
This should have `DEFAULT 1` if you force it to not be `NULL` cause in tables that contain data it will fail IIRC.
```suggestion
        $this->addSql('ALTER TABLE sylius_product_variant ADD enabled TINYINT(1) DEFAULT 1 NOT NULL');
```

Reply:
Yeah you're right we just copied what has been done for other `enabled` columns in the core but to avoid BC break we agree that the default value should be provided here. We just committed your suggestion.

Reply:
Sorry but I just realized that this needs column definition in Doctrine mapping too. And as stated here https://stackoverflow.com/questions/3376881/default-value-in-doctrine Doctrine column definition mapping isn't portable across various DBMS.

The default value of 1 (true) is already set at PHP object level so I don't know if it's really needed.

@pamil what do you think?

Reply:
Second option always used in legacy systems is:
- add field without `NOT NULL`,
- set default value on all items in that table where value `IS NULL`,
- add `ALTER TABLE` with proper `NOT NULL`,
So you need 3 queries, but you avoid adding custom config into field settings

Reply:
What if we simply add the following query to the migration (after the `ALTER TABLE`)?

```sql
$this->addSql('UPDATE sylius_product_variant SET enabled=1');
```

Reply:
It will not work with setting value that was forced to be set by `NOT NULL`, it will fail already on that query as mentioned.
```suggestion
        $this->addSql('ALTER TABLE sylius_product_variant ADD enabled TINYINT(1) NULL');
        $this->addSql('UPDATE sylius_product_variant SET enabled = 1');
        $this->addSql('ALTER TABLE sylius_product_variant ADD enabled TINYINT(1) NOT NULL');
```

Reply:
Ok I'd also like to have a core team member opinion here.

Reply:
Btw it's not true that it doesn't work with existing variants. I tested on MySQL and MariaDB, the statement:

```sql
ALTER TABLE sylius_product_variant ADD enabled TINYINT(1) NOT NULL
```

Will work even with existing variants.

Reply:
Added the following statement to the migration:

```sql
UPDATE sylius_product_variant SET enabled = 1
```
I'm going to resolve this.